storage/chunked_zstd.go

```diff
@@ -0,0 +1,834 @@
+// +build !containers_image_storage_stub
+
+package storage
+
+import (
+	archivetar "archive/tar"
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime"
+	"mime/multipart"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/pkg/compression"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage/drivers"
+	driversCopy "github.com/containers/storage/drivers/copy"
+	"github.com/containers/storage/pkg/archive"
+	"github.com/containers/storage/pkg/idtools"
+	"github.com/containers/storage/pkg/system"
+	securejoin "github.com/cyphar/filepath-securejoin"
+	"github.com/klauspost/compress/zstd"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/vbatts/tar-split/archive/tar"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	maxNumberMissingChunks = 1024
+	newFileFlags           = (unix.O_CREAT | unix.O_TRUNC | unix.O_WRONLY | unix.O_EXCL)
+)
+
+type DirectDiffInput struct {
+	stream         types.ImageSourceSeekable
+	store          *storageImageDestination
+	manifest       []byte
+	ctx            context.Context
+	srcInfo        types.BlobInfo
+	layersMetadata map[string][]compression.FileMetadata
```

Comment:
All of the fields, and especially the maps/arrays, will eventually need documentation.